src/node_url_pattern.cc

```diff
@@ -726,61 +631,16 @@ static void Initialize(Local<Object> target,
   auto signature = Signature::New(isolate, ctor_tmpl);
 
   instance_template->SetInternalFieldCount(URLPattern::kInternalFieldCount);
-  prototype_template->SetAccessorProperty(
-      env->protocol_string(),
-      FunctionTemplate::New(
-          isolate, URLPattern::Protocol, Local<Value>(), signature),
-      Local<FunctionTemplate>(),
-      attributes);
-
-  prototype_template->SetAccessorProperty(
-      env->username_string(),
-      FunctionTemplate::New(
-          isolate, URLPattern::Username, Local<Value>(), signature),
-      Local<FunctionTemplate>(),
-      attributes);
-
-  prototype_template->SetAccessorProperty(
-      env->password_string(),
-      FunctionTemplate::New(
-          isolate, URLPattern::Password, Local<Value>(), signature),
-      Local<FunctionTemplate>(),
-      attributes);
-
-  prototype_template->SetAccessorProperty(
-      env->hostname_string(),
-      FunctionTemplate::New(
-          isolate, URLPattern::Hostname, Local<Value>(), signature),
-      Local<FunctionTemplate>(),
-      attributes);
-
-  prototype_template->SetAccessorProperty(
-      env->port_string(),
-      FunctionTemplate::New(
-          isolate, URLPattern::Port, Local<Value>(), signature),
-      Local<FunctionTemplate>(),
-      attributes);
 
-  prototype_template->SetAccessorProperty(
-      env->pathname_string(),
-      FunctionTemplate::New(
-          isolate, URLPattern::Pathname, Local<Value>(), signature),
-      Local<FunctionTemplate>(),
-      attributes);
-
-  prototype_template->SetAccessorProperty(
-      env->search_string(),
-      FunctionTemplate::New(
-          isolate, URLPattern::Search, Local<Value>(), signature),
-      Local<FunctionTemplate>(),
-      attributes);
-
-  prototype_template->SetAccessorProperty(
-      env->hash_string(),
-      FunctionTemplate::New(
-          isolate, URLPattern::Hash, Local<Value>(), signature),
-      Local<FunctionTemplate>(),
+#define URL_PATTERN_COMPONENT_GETTERS(uppercase_name, _)                       \
+  prototype_template->SetAccessorProperty(                                     \
+      env->username_string(),                                                  \
```

Comment:
This needs to change

```suggestion
      env->##lowercase_name##_string(),                                                  \
```